gtsam/hybrid/tests/testHybridGaussianFactor.cpp

```diff
@@ -54,32 +54,49 @@ TEST(HybridGaussianFactor, Constructor) {
   CHECK(it == factor.end());
 }
 
+/* ************************************************************************* */
+namespace testA {
+DiscreteKey m1(1, 2);
+
+auto A1 = Matrix::Zero(2, 1);
+auto A2 = Matrix::Zero(2, 2);
+auto b = Matrix::Zero(2, 1);
+
+auto f10 = std::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
+auto f11 = std::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
+}  // namespace testA
+
+/* ************************************************************************* */
+// Test simple to complex constructors...
+TEST(HybridGaussianFactor, ConstructorVariants) {
+  using namespace testA;
+  HybridGaussianFactor fromFactors({X(1), X(2)}, m1, {f10, f11});
+
+  std::vector<GaussianFactorValuePair> pairs{{f10, 0.0}, {f11, 0.0}};
+  HybridGaussianFactor fromPairs({X(1), X(2)}, {m1}, pairs);
+  assert_equal(fromFactors, fromPairs);
+
+  HybridGaussianFactor::FactorValuePairs decisionTree({m1}, pairs);
+  HybridGaussianFactor fromDecisionTree({X(1), X(2)}, {m1}, decisionTree);
+  assert_equal(fromDecisionTree, fromPairs);
+}
+
 /* ************************************************************************* */
 // "Add" two hybrid factors together.
 TEST(HybridGaussianFactor, Sum) {
-  DiscreteKey m1(1, 2), m2(2, 3);
+  using namespace testA;
+  DiscreteKey m2(2, 3);
 
-  auto A1 = Matrix::Zero(2, 1);
-  auto A2 = Matrix::Zero(2, 2);
   auto A3 = Matrix::Zero(2, 3);
-  auto b = Matrix::Zero(2, 1);
-  Vector2 sigmas;
-  sigmas << 1, 2;
-
-  auto f10 = std::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
-  auto f11 = std::make_shared<JacobianFactor>(X(1), A1, X(2), A2, b);
   auto f20 = std::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
   auto f21 = std::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
   auto f22 = std::make_shared<JacobianFactor>(X(1), A1, X(3), A3, b);
-  std::vector<GaussianFactorValuePair> factorsA{{f10, 0.0}, {f11, 0.0}};
-  std::vector<GaussianFactorValuePair> factorsB{
-      {f20, 0.0}, {f21, 0.0}, {f22, 0.0}};
 
   // TODO(Frank): why specify keys at all? And: keys in factor should be *all*
```

Comment:
We should talk about this TODO. I guess the compexity here is getting the discrete keys from the DecisionTree since there is no way to do it currently.

Reply:
Right, let's chat